diff --git a/ios/PickerView.swift b/ios/PickerView.swift
index 18d40e00fa6ad673bf13e566c8b4dfe7d02dafb3..0404705f16e42854c4112e14c5ba224429468fe6 100644
--- a/ios/PickerView.swift
+++ b/ios/PickerView.swift
@@ -28,38 +28,42 @@ struct PickerView: ExpoSwiftUI.View {
   }
 
   var body: some View {
-    if #available(iOS 17.0, tvOS 17.0, *) {
-      Picker(props.label ?? "", selection: $selection) {
-        ForEach(Array(props.options.enumerated()), id: \.0) { index, option in
-          Text(option).tag(index)
-        }
+    Picker(props.label ?? "", selection: $selection) {
+      ForEach(Array(props.options.enumerated()), id: \.0) { index, option in
+        Text(option).tag(index)
       }
-      .tint(props.color)
-      .modifier(CommonViewModifiers(props: props))
+    }
+    .tint(props.color)
+    .modifier(CommonViewModifiers(props: props))
 #if !os(tvOS)
-      .if(props.variant == "wheel", { $0.pickerStyle(.wheel) })
-      .if(props.variant == "palette", { $0.pickerStyle(.palette) })
-#endif
-      .if(props.variant == "segmented", { $0.pickerStyle(.segmented) })
-      .if(props.variant == "inline", { $0.pickerStyle(.inline) })
-      .if(props.variant == "menu", { $0.pickerStyle(.menu) })
-      .onChange(of: selection, perform: { newValue in
-        if props.selectedIndex == newValue {
-          return
-        }
-        let payload = [
-          "index": newValue,
-          "label": props.options[newValue]
-        ]
-        props.onOptionSelected(payload)
-      })
-      .onReceive(props.selectedIndex.publisher, perform: { newValue in
-        if prevSelectedIndex == newValue {
-          return
-        }
-        selection = newValue
-        prevSelectedIndex = newValue
-      })
+    .if(props.variant == "wheel", { $0.pickerStyle(.wheel) })
+    .apply { view in
+      if #available(iOS 17.0, *), props.variant == "palette" {
+        view.pickerStyle(.palette)
+      } else {
+        view
+      }
     }
+#endif
+    .if(props.variant == "segmented", { $0.pickerStyle(.segmented) })
+    .if(props.variant == "inline", { $0.pickerStyle(.inline) })
+    .if(props.variant == "menu", { $0.pickerStyle(.menu) })
+    .onChange(of: selection, perform: { newValue in
+      if props.selectedIndex == newValue {
+        return
+      }
+      let payload = [
+        "index": newValue,
+        "label": props.options[newValue]
+      ]
+      props.onOptionSelected(payload)
+    })
+    .onReceive(props.selectedIndex.publisher, perform: { newValue in
+      if prevSelectedIndex == newValue {
+        return
+      }
+      selection = newValue
+      prevSelectedIndex = newValue
+    })
   }
 }
diff --git a/node_modules/@expo/ui/ios/View+Extensions.swift b/ios/View+Extensions.swift
new file mode 100644
index 0000000000000000000000000000000000000000..943803ac39943a9bf594dbdcbaf605024d5a9f69
--- /dev/null
+++ b/ios/View+Extensions.swift
@@ -0,0 +1,7 @@
+// Copyright 2025-present 650 Industries. All rights reserved.
+
+import SwiftUI
+
+extension View {
+  func apply<V: View>(@ViewBuilder _ block: (Self) -> V) -> V { block(self) }
+}
